app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,50 @@
+class AccountActivationsController < ApplicationController
+
+before_action :get_user, only: [:edit, :create]
+before_action :check_expiration, only: [:edit]
+  def show
+    render :activate
+  end 
+
+  def edit
+    user = User.find_by(email: params[:email].downcase)
+    if user && !user.activated? && @user.authenticated?(:activation, params[:id])
+        user.activate
+        log_in user
+        flash[:success] = t('.account_activated')
+        redirect_to root_url
+    else
+      flash[:danger] = t('.invalid_activation_link')
+      redirect_to root_url
+    end
+  end
+
+  # Create a new activation link
+  def create
+    user = User.find_by(email: params[:email].downcase)
+    if user && !user.activated?
+      user.update_attribute(:activation_token, User.secure_token)
+      user.update_attribute(:activation_digest, User.digest(user.activation_token))
```

Comment:
I would replace this with:

```ruby
activation_token = User.secure_token
user.update_attributes(
  activate_token: activation_token,
  activation_digest: User.digest(activation_token)
)
```

That way you're only updating the user once instead of twice.

Another way you could do it is:

```ruby
user.activation_token = User.secure_token
user.activation_digest = User.digest(user.activation_token)
user.save
```